types/share_proof.go

```diff
@@ -91,18 +90,19 @@ func (sp ShareProof) Validate(root []byte) error {
 		}
 	}
 
-	if err := sp.RowProof.Validate(root); err != nil {
+	if err := sp.RowProof.Validate(); err != nil {
 		return err
 	}
 
-	if ok := sp.VerifyProof(); !ok {
-		return errors.New("share proof failed to verify")
-	}
-
 	return nil
 }
 
-func (sp ShareProof) VerifyProof() bool {
+// VerifyProof check if the proof is valid then
+// verify that the share to data root proof is valid.
+func (sp ShareProof) VerifyProof(root []byte) bool {
+	if err := sp.Validate(); err != nil {
+		return false
+	}
```

Comment:
[separate issue] since `VerifyProof` only returns a bool value, I expect callers will be need debugging tools if they receive a `false` here.

We could consider modifying the API so that an error is returned if VerifyProof returns false.

```go
// VerifyProof returns true if the proof is valid and the share to data root proof is valid.
// VerifyProof returns false and an error if ...
func (sp ShareProof) VerifyProof(root []byte) (bool, error) {
```

Reply:
https://github.com/celestiaorg/celestia-core/issues/1359